fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,195 @@
+            if not _grpc_interface_file_dir:
+                raise ValueError("'interface_dir' is not defined in the app.config file")
+
+            log.info("gRPC Package Name : {} gRPC Stub Method Name : {} gRPC Request Class Name : {}" \
+                     .format(_grpc_package_name, _grpc_rpc_stub_method_name, _grpc_request_method_name))
+
+
+            # Importing gRPC proto buff files from the given interface directory.
+            try:
+                module_files = grpc_helper_obj.get_interface_file_names(_grpc_interface_file_dir, _grpc_package_name)
+
+                # Removing __pycache__ directory from module list since it's not required
+                if "__pycache__" in module_files:
+                    pycache_index = module_files.index("__pycache__")
+                    module_files.pop(pycache_index)
+
+                if not module_files:
+                    raise ValueError("No gRPC protocol buffer files found in your interface_dir for the service '{0}'. Please copy the {0}_pb2.py and {0}_pb2_grpc.py files to your interface_dir".format(_grpc_package_name))
+                else:
+                    for file_name in module_files:
+                        if not (file_name.endswith('.py') or file_name.endswith('.pyc')) and 'pb' not in file_name:
+                            raise ValueError("Invalid file '{0}' found in your interface_dir for the service '{1}'. Please only copy the {1}_pb2.py and {1}_pb2_grpc.py files to your interface_dir".format(file_name, _grpc_package_name))
+
+                grpc_interface_module_list = grpc_helper_obj.get_grpc_interface_module(module_files,
+                                                                                       _grpc_interface_file_dir,
+                                                                                       _grpc_package_name)
+                log.debug(grpc_interface_module_list)
+
+                grpc_stub_tuple = grpc_helper_obj.get_grpc_class(grpc_interface_module_list, 'stub')
+                if not grpc_stub_tuple:
+                    raise ValueError("No gRPC 'stub' class found in the protocol buffer files in the interface_dir")
+
+                grpc_request_tuple = grpc_helper_obj.get_grpc_class(grpc_interface_module_list,
+                                                                    _grpc_request_method_name.lower())
+            except Exception as error_msg:
+                raise ValueError("Failed to load the gRPC protocol buffer files:\nERROR: {0}".format(error_msg))
+
+            # Converting resilient function input data into json object
+            try:
+                grpc_function_json_data = json.loads(grpc_function_data)
+            except Exception:
+                raise FunctionError(u"Function input 'grpc_function_data' must be a JSON formatted String: {0}".format(grpc_function_json_data))
+
+            # Checking Communication type
+            if _grpc_communication_type.lower().strip() == "unary":
```

Comment:
This is a very static way to check for the communication types we support. It would be modular to do something like: (which will make this code easier to extend in the future):
```
 SUPPORTED_COMM_METHODS = ["unary"]

        if _grpc_communication_type not in SUPPORTED_COMM_METHODS:
            raise ValueError("{0} is not a supported Communication Type. Supported types are: {1}".format(_grpc_communication_type, SUPPORTED_COMM_METHODS))

```

Reply:
Then you can remove the else block on line 165:
```
            else:
                raise NotImplementedError(
                    "Not Implemented these {} communication types ".format(_grpc_communication_type))
```

Reply:
ok